share/ipld/get.go

```diff
@@ -215,41 +205,23 @@ func GetLeavesByNamespace(
 		retrievalErr error
 	)
 
-	// we overallocate space for leaves since we do not know how many we will find
-	// on the level above, the length of the Row is passed in as maxShares
-	leaves := make([]ipld.Node, maxShares)
-
-	// if non-nil proof container provided, collect proofs while traversing the tree and fill put them
-	// into container after
-	var collectProofs = proofContainer != nil
-	var proofs *proofCollector
-	if collectProofs {
-		proofs = newProofCollector(maxShares)
-	}
-
 	for {
 		var j *job
 		var ok bool
 		select {
 		case j, ok = <-jobs:
 		case <-ctx.Done():
-			return nil, ctx.Err()
+			return ctx.Err()
 		}
 
 		if !ok {
 			// if there were no leaves under the given root in the given namespace,
-			// both return values are nil. otherwise, the error will also be non-nil.
-			if bounds.lowest == int64(maxShares) {
-				return nil, retrievalErr
-			}
-
-			if collectProofs {
-				proofContainer.Start = int(bounds.lowest)
-				proofContainer.End = int(bounds.highest) + 1
-				proofContainer.Nodes = proofs.Nodes()
+			// leaves and error will be nil. otherwise, the error will also be non-nil.
+			if data.noLeaves() {
+				return retrievalErr
 			}
 
-			return leaves[bounds.lowest : bounds.highest+1], retrievalErr
+			return retrievalErr
```

Comment:
Looks like you get rid of if statement here. Anyway it is `retrievalErr` returned

Reply:
Yeap. nice catch. Thank you